lonestar/analytics/gpu/pointstoanalysis/andersen.cu

```diff
@@ -34,6 +34,32 @@
 
 using namespace thrust;
 
+#if (__CUDACC_VER_MAJOR__ >= 9 && defined(__CUDA_ARCH__) &&  __CUDA_ARCH__ >= 300) 
+#define BALLOT_INST(x) {      \
+  __ballot_sync(0xffffffff,x) \
```

Comment:
Isn't this available in previous versions? I thought it was backward-compatible.

Reply:
Moreover, why is this specific to PTA? We use ballot, any, all instructions for distributed GPU apps as well.

Check libgpu\include\cutil_subset.h and lonestar\analytics\distributred\bfs\bfs_push_cuda.cu

We should have a consistent solution provided in libgpu, which is used in all apps.

Reply:
@chenxuhao this needs to be fixed before we can merge.

Reply:
This is what's in libgpu\include\cutil_subset.h 

unsigned int active = __ballot_sync(0xffffffff, 1);

What do you mean about it?

Reply:
There are 2 issues with this code:

(1) We are using *_sync funcs in other places, so why a special #if (and else) only for PTA? We should use it consistently everywhere *_sync is used.

(2) More importantly, I do not think #if (and else) is required. We require gcc 7 or above, which implies we require CUDA 9 or above. The first two conditions in #if will always to true. The last condition I'm not sure. I thought *_sync was backward compatible for different for earlier CUDA capabilities. I do not see anything in the CUDA programming guide which says we need to use a different function for earlier CUDA capabilities. Why is the else part required?

Reply:
I will remove else part

Reply:
Thanks! Then there's no need to use #define. Directly use *_sync functions everywhere.

Reply:
removed